### PR TITLE
fix: CNS should ignore NNCs that are being deleted

### PR DESCRIPTION
**Pull request by rbtr**
**Reason for Change**:

If an NNC is marked for deletion, CNS should not use it to reinitialize state or ingest any events for it.

**Issue Fixed**:

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] relevant PR labels added

**Notes**:

### CODE REVIEW

**Review by nddq — APPROVED** [review]
lgtm

File: cns/service/main.go [hunk 1]
```diff
@@ -1201,6 +1201,9 @@ func reconcileInitialCNSState(ctx context.Context, cli nodeNetworkConfigGetter,
 	}
 
 	logger.Printf("Retrieved NNC: %+v", nnc)
+	if !nnc.DeletionTimestamp.IsZero() {
```
Comment (tyler-lloyd) [inline cns/service/main.go:1204]: assuming there no weird kubernetes thing that sets a deletiontimestamp to `January 1, year 1, 00:00:00 UTC` just to make it non-nil to signal deletion?

Comment (rbtr) [reply cns/service/main.go:1204]: The timestamp is set on objects with finalizers to indicate that they are pending deletion to those things that own the finalizers. I don't think in practice that this is ever non-nil and zero

**Review by thatmattlong — DISMISSED** [review]

**Review by nddq — APPROVED** [review]

**Pull request merged**